packages/next/build/output/index.ts

```diff
@@ -5,7 +5,7 @@ import createStore from 'next/dist/compiled/unistore'
 import formatWebpackMessages from '../../client/dev/error-overlay/format-webpack-messages'
 import { OutputState, store as consoleStore } from './store'
 
-export function startedDevelopmentServer(appUrl: string) {
+export function startedDevelopmentServer(appUrl: string | null) {
```

Comment:
Is `appUrl` supposed to be where the app is listening or where the rest of the world can find it? Doesn't this have implications for if the app is running behind a reverse proxy? With either socket or port based hosting, it's common to have the publicly accessible URL be different.

Reply:
It's used in logging output only ("started server on http://..."), so I assumed it would not be an issue. I also don't see any other code accessing the store, outside `packages/next/build/output`. I guess it could be an issue if something were to parse the log message, but then is that considered a stable interface of Next.js?

The store already allowed `null` here, I simply modified this setter so that you can (re)set the store to `null` with it. If it is `null`, the log message will simply be skipped.

In either case, I also assumed this'd be a scenario where the user kind of knows what they're doing. 🙃

Reply:
My assumption was that the store *might* like the base app URL for routing purposes. Relative links would not care. I guess it comes down to what is the intention of the `appUrl` value in the store.

Reply:
I don't believe it's intended for routing. The store was added in 56b1f81, and looks to be only there for console output. It's also aliased as `consoleStore` in places.

Reply:
Well, then I guess this is irrelevant!

Reply:
Maybe it's an idea to rename it something that makes this more clear? Maybe `logUrl`?